core/lib/makeswift/components/button-link/button-link.makeswift.tsx

```diff
@@ -0,0 +1,60 @@
+import { MakeswiftComponentType } from '@makeswift/runtime';
+import { Link, Select, Style, TextInput } from '@makeswift/runtime/controls';
+
+import { ButtonLink, Props } from '@/vibes/soul/primitives/button-link';
+import { runtime } from '~/lib/makeswift/runtime';
+
+type MSButtonLinkProps = Omit<Props, 'href'> & {
+  link: { href?: string; target?: string };
+  text: string;
+};
+
+runtime.registerComponent(
+  function MSButtonLink({ link, text, ...props }: MSButtonLinkProps) {
+    return (
+      <ButtonLink href={link.href ?? ''} target={link.target ?? ''} {...props}>
```

Comment:
```suggestion
      <ButtonLink href={link.href ?? '#'} target={link.target} {...props}>
```

We don't need to pass `target`, right? If we do, let's do `_self`